packages/@expo/fingerprint/src/Fingerprint.types.ts

```diff
@@ -150,10 +155,50 @@ export type Config = Pick<
   | 'enableReactImportsPatcher'
   | 'useRNCoreAutolinkingFromExpo'
   | 'debug'
+  | 'fileHookTransform'
 > & {
   sourceSkips?: SourceSkips | SourceSkipsKeys[];
 };
 
+/**
+ * Hook function to transform file content sources before hashing.
+ */
+export type FileHookTransformFunction = (
+  /**
+   * Source from HashSourceFile or HashSourceContents.
+   */
+  source: FileHookTransformSource,
+
+  /**
+   * The chunk of file content.
+   * When the stream reaches the end, the chunk will be null.
+   */
+  chunk: Buffer | string | null,
+
+  /**
+   * Indicates the end of the file.
+   */
+  isEndOfFile: boolean,
+
+  /**
+   * The encoding of the chunk.
+   */
+  encoding: BufferEncoding
+) => Buffer | string | null;
+
+/**
+ * The `source` parameter for `FileHookTransformFunction`.
+ */
+export type FileHookTransformSource =
+  | {
+      type: 'file';
+      filePath: string;
+    }
+  | {
+      type: 'contents';
+      id: string;
+    };
+
 //#region internal types
 
 export type NormalizedOptions = Omit<Options, 'ignorePaths'> & {
```

Comment:
Think we should include anything in `DebugInfo` about transformations? My gut says no, but I wonder if there's a use case.

Reply:
good point! i try to add a `debugInfo.isTransformed` info in https://github.com/expo/expo/pull/33610/commits/3de36023f5c12df6773904fd131a5b17b02d0a12. hopes that would be good enough.